libraries/manipulator.rb

```diff
@@ -208,7 +208,9 @@ def new_content
     entries = hostsfile_header
     entries += unique_entries.map(&:to_line)
     entries << ''
-    entries.join("\n")
+    line_ending = "\n"
+    line_ending = "\r\n" if node['platform_family'] == 'windows'
```

Comment:
We could combine this line (ex. `line_ending = node['platform_family'] == 'windows' ? "\r\n" : "\n"`) so that we don't assign line_ending twice, in the uncommon case.